src/index.js

```diff
@@ -17,12 +17,11 @@ export function asyncActionCreator(type, config) {
   if (typeof config === 'function') config = {action: config}; // eslint-disable-line no-param-reassign
   const {client, server, schema} = config;
 
-  return payload => (...args) => {
-    const dispatch = args[0];
+  return payload => (dispatch, ...args) => {
     const action = config.action || (isNode ? server : client);
     args.unshift(payload);
```

Comment:
`args.unshift(payload);` still needs to go; at the moment it will double up the `payload` parameter in the arguments list